### ISSUE TITLE
Feature request for making dynamic gradient clipping 

### ISSUE THREAD

**Issue by kweisamx** [tensorflow/tensorflow#14566]
That I know about gradient, tf has 
https://www.tensorflow.org/versions/r0.12/api_docs/python/train/gradient_clipping

But I need the clipping gradient can change at training, and I see the all clip gradient doc, not found this function.

- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: python pip 
- **TensorFlow version (use command below)**: v 1.3.0
- **Python version**: 3.5.2
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:5.4.0
- **CUDA/cuDNN version**:Cuda compilation tools, release 8.0, V8.0.61
- **GPU model and memory**: 1080Ti 8G
- **Exact command to reproduce**:

Thanks

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by kweisamx** [comment]
Hi @tensorflowbutler 
I update the infomation
thx

**Comment by drpngx** [comment]
What do you mean by dynamic?

**Comment by kweisamx** [comment]
like the learning rate can decay by the step , because the clipping is fixed when begining, 
maybe the clipping can adjust when training like learning

**Comment by drpngx** [comment]
Sounds good. Do you want to send a PR?

**Comment by ppwwyyxx** [comment]
@kweisamx You can clip by a tensor which changes with the steps. Why do you think the clipping value __has to be__ fixed at the beginning?

**Comment by kweisamx** [comment]
@ppwwyyxx yep, that i have tried, but the clipping not changing by step,

**Comment by ppwwyyxx** [comment]
Then it sounds more like a bug report and you'd better post your code so people can find out the problems. From the documentation of `tf.clip_by_*` this feature is definitely supported.

**Comment by HughKu** [comment]
Hi @kweisamx, any update on this issue?

I test a minimal example for `tf.clip_by_value` and it works fine.
```python
import tensorflow as tf

tf.reset_default_graph()

W = tf.get_variable('W', shape=[2, 4], dtype=np.float32)
clip_min_tf = tf.get_variable("clip_min_tf", shape=[], dtype=tf.float32)
clip_max_tf = tf.get_variable("clip_max_tf", shape=[], dtype=tf.float32)

op = tf.clip_by_value(W, clip_min_tf, clip_max_tf)

init = tf.global_variables_initializer()

with tf.Session() as sess:
    sess.run(init)
    
    sess.run(tf.assign(W, [[1, 1, 2, 4], [3, 4, 8, 5]]))
    
    sess.run(tf.assign(clip_min_tf, 2))
    sess.run(tf.assign(clip_max_tf, 5))
    sess.run(op)
    print(op.eval())
    # >> [[2. 2. 2. 4.], [3. 4. 5. 5.]]

    sess.run(tf.assign(clip_min_tf, 4))
    sess.run(tf.assign(clip_max_tf, 7))
    sess.run(op)
    print(op.eval())
    # >> [[4. 4. 4. 4.], [4. 4. 7. 5.]]
```
